kafka/pykafka/balancedconsumer.py

```diff
@@ -0,0 +1,371 @@
+        :type fetch_wait_max_ms: int
+        :param refresh_leader_backoff_ms: backoff time to refresh the leader of
+            a partition after it loses the current leader
+        :type refresh_leader_backoff_ms: int
+        :param offsets_channel_backoff_ms: backoff time to retry offset
+            commits/fetches
+        :type offsets_channel_backoff_ms: int
+        :param offsets_commit_max_retries: Retry the offset commit up to this
+            many times on failure.
+        :type offsets_commit_max_retries: int
+        :param auto_offset_reset: what to do if an offset is out of range
+        :type auto_offset_reset: int
+        :param consumer_timeout_ms: throw a timeout exception to the consumer
+            if no message is available for consumption after the specified
+            interval
+        :type consumer_timeout_ms: int
+        :param rebalance_max_retries: Maximum number of attempts before failing to
+            rebalance
+        :type rebalance_max_retries: int
+        :param rebalance_backoff_ms: Backoff time between retries during rebalance.
+        :type rebalance_backoff_ms: int
+        :param zookeeper_connection_timeout_ms: The max time that the client
+            waits while establishing a connection to zookeeper.
+        :type zookeeper_connection_timeout_ms: int
+        :param zookeeper_connect: comma separated ip:port strings of the
+            zookeeper nodes to connect to
+        :type zookeeper_connect: str
+        """
+        if not isinstance(cluster, weakref.ProxyType):
+            self._cluster = weakref.proxy(cluster)
+        else:
+            self._cluster = cluster
+        self._consumer_group = consumer_group
+        self._topic = topic
+
+        self._auto_commit_enable = auto_commit_enable
+        self._auto_commit_interval_ms = auto_commit_interval_ms
+        self._fetch_message_max_bytes = fetch_message_max_bytes
+        self._fetch_min_bytes = fetch_min_bytes
+        self._rebalance_max_retries = rebalance_max_retries
+        self._num_consumer_fetchers = num_consumer_fetchers
+        self._queued_max_messages = queued_max_messages
+        self._fetch_wait_max_ms = fetch_wait_max_ms
+        self._rebalance_backoff_ms = rebalance_backoff_ms
+        self._consumer_timeout_ms = consumer_timeout_ms
+
+        self._consumer = None
+        self._consumer_id = "{}:{}".format(socket.gethostname(), uuid4())
+        self._partitions = set()
+        self._setting_watches = True
+
+        self._topic_path = '/consumers/{}/owners/{}'.format(self._consumer_group,
+                                                            self._topic.name)
+        self._consumer_id_path = '/consumers/{}/ids'.format(self._consumer_group)
+
+        def _close_zk_connection(signum, frame):
+            self._zookeeper.stop()
+            self._consumer.stop()
+            sys.exit()
+        signal.signal(signal.SIGINT, _close_zk_connection)
```

Comment:
Yes, I suppose this is more of a command line use case. I'll move the signal-catching to my test script.